### PR TITLE
More range updates

### PR DESCRIPTION
**Pull request by kmharrington**
More functionality for Ranges and RangesMatrix. 
1) Addition for RangesMatrix
2) `buffer` and `buffered` for expanding Ranges by set amounts
3) `close_gaps` to combine any ranges within a set amount

### CODE REVIEW

File: src/Ranges.cxx [hunk 1]
```diff
@@ -120,6 +120,62 @@ Ranges<T>& Ranges<T>::merge(const Ranges<T> &src)
     return *this;
 }
 
+//Buffer Range in place
+template <typename T>
+Ranges<T>& Ranges<T>::buffer(const T buff)
+{
+    auto p0 = this->segments.begin();
+    while (p0 != this->segments.end()){
```
Comment (mhasself) [inline src/Ranges.cxx:128]: Throughout... can you add the space between `){` to make it `) {`?

File: src/Ranges.cxx [hunk 2]
```diff
@@ -672,6 +728,17 @@ using namespace boost::python;
          return_internal_reference<>(),                                 \
          args("self", "src"),                                           \
          "Merge ranges from another " #CLASSNAME " into this one.")     \
+    .def("buffer", &CLASSNAME::buffer,                                  \
+        return_internal_reference<>(),                                  \
+        args("self", "buff"),                                           \
+        "Buffer each interval by an amount specificed by buff")         \
```
Comment (mhasself) [inline src/Ranges.cxx:753]: speeling

**Comment by kmharrington** [comment]
Done! Also I added ~2 more functions to `RangesMatrix`

**Comment by mhasself** [comment]
Fun... can I advocate for renaming to zeros_like and ones_like?  I think that's more consistent with numpy terminology (where empty means unitialized).  Also to avoid another comment, you need a line between your two new functions in RangesMatrix!

**Comment by kmharrington** [comment]
Makes sense, updated!

**Pull request merged**